packages/rum-recorder/src/domain/segmentCollection.ts

```diff
@@ -33,32 +33,56 @@ export const MAX_SEGMENT_DURATION = 30_000
 // To help investigate session replays issues, each segment is created with a "creation reason",
 // indicating why the session has been created.
 
+let workerSingleton: DeflateWorker
+
 export function startSegmentCollection(
   lifeCycle: LifeCycle,
   applicationId: string,
   session: RumSession,
   parentContexts: ParentContexts,
   send: (data: Uint8Array, meta: SegmentMeta) => void
 ) {
-  const worker = createDeflateWorker()
+  if (!workerSingleton) {
+    workerSingleton = createDeflateWorker()
+  }
   return doStartSegmentCollection(
     lifeCycle,
     () => computeSegmentContext(applicationId, session, parentContexts),
     send,
-    worker
+    workerSingleton
   )
 }
 
+const enum StateType {
+  WaitingForInitialRecord,
+  SegmentPending,
+  Stopped,
+}
+type State =
```

Comment:
I'd be in favor of having more explicit names for State/StateType, what about something like SegmentCollectionState/SegmentCollectionStatus?